Automattic/wp-calypso
# Replace design screenshots with mShots in New Onboarding

#### Changes proposed in this Pull Request

* Generate an mShots screenshot for the design preview that passes in a locale to the `template/demo` endpoint
* Add an `MshotsImage` component that can deal with hiding the mShots default loading gif, and continue to request the image until it loads the final screenshot. The user should only see a pulsing rectangle until the image finally loads

#### Outstanding issues and todos

* [ ] The vertically oriented screenshots aren't _quite_ how we'd like them, some of the viewport height based elements appear stretched vertically a little bit more than desired. This may or may not be a deal-breaker.
* [ ] Currently the image file size is a bit too much greater than the existing webp approach. Separately to this PR we're proposing adding in a quality param for mShots to support smaller file sizes.
* [ ] The logic to keep the image hidden until _really_ loaded is a bit hacky and relies on a hard-coded URL. Is this safe enough, and does it hold up to testing when generating new screenshots? Is there a reliable way to test this, or write a test for it?
* [ ] Are we caching correctly? If a user goes back and forward between steps, does the browser re-download the images, or does it correctly cache them? We should ensure that the images aren't being re-downloaded multiple times.

#### Screenshot

![mshots-in-design-step-small](https://user-images.githubusercontent.com/14988353/98332307-883c6a80-2052-11eb-89dd-ff332c0ca712.gif)

#### Testing instructions



* Go to Calypso live URL below and append the mshots flag to the end of the URL, e.g. add `?flags=gutenboarding/mshot-preview` or build locally and run `calypso.localhost:3000/new/ko?flags=gutenboarding/mshot-preview`.
* Either set your user language, or add a locale to the URL and go to Gutenboarding, e.g. `/new/ko?flags=gutenboarding/mshot-preview` or `/new/de?flags=gutenboarding/mshot-preview`
* At the design step, the previews should be translated
* Test this in a number of different languages
* Test this with limited Network speed (e.g. Fast 3G or Slow 3G in the Network tab in dev tools) — this will demonstrate that the file sizes for the images are currently too big.

Related to #37665 
Fixes #48867
